pkg/debug/spec.sh

```diff
@@ -37,9 +59,83 @@ pci_iommu_group() {
 # $1 is the name; $2 is the PciLong value
 get_assignmentgroup() {
     local pcilong=$2
-    pci_iommu_group "$pcilong"
+    local grp
+    grp=$(pci_iommu_group "$pcilong")
+    if pci_iommugroup_includes_unknown "${pcilong}" "${grp}"; then
+        echo ""
+    else
+        echo "group${grp}"
+    fi
 }
 
+# pci_iommugroup_includes_unknown($PCIID, $IOMMUGRPNUM)
+# returns whether or not there is some unknown to EVE-OS type of
+# device in the group
+pci_iommugroup_includes_unknown() {
+    local pcilong=$1
+    local iommugrpnum=$2
+    local grp
+    local pci
+    local ztype
+#shellcheck disable=SC2044
+    for a in $(find /sys/kernel/iommu_groups/ -type l); do
+        grp=$(echo "${a}" | awk -F/ '{print $5}')
+        pci=$(echo "${a}" | awk -F/ '{print $7}')
+        [ "${grp}" = "${iommugrpnum}" ] || continue
+        [ "${pci}" != "${pcilong}" ] || continue
+
+        # Check if $pci is of unknown type
+        ztype=$(pci_to_ztype "$pci")
+        if [ "$ztype" == 255 ]; then
+            return 0
+        fi
+    done
+    return 1
+}
+
+# pci_to_ztype($PCI_ID) returns a numeric ztype
+pci_to_ztype() {
+    local pci=$1
+    if [ -d "/sys/bus/pci/devices/${pci}/net" ]; then
+        local ifname
+        local ztype
+        ifname=$(ls "/sys/bus/pci/devices/${pci}/net")
+        ztype=1
+        if [ "${ifname:0:4}" = "wlan" ]; then
+            ztype=5
+        elif [ "${ifname:0:4}" = "wwan" ]; then
+            ztype=6
+        fi
+    elif lspci -D -s "${pci}" | grep -q USB; then
```

Comment:
This hardcoding of USB, Audio, VGA seems a bit suspect.

Reply:
The script already has those strings hardcoded elsewhere.